agent/wptdriver/webpagetest.cc

```diff
@@ -161,6 +186,57 @@ bool WebPagetest::GetTest(WptTestDriver& test) {
 }
 
 /*-----------------------------------------------------------------------------
+Call just before rebooting. If return false, reboot is cancelled
+-----------------------------------------------------------------------------*/
+bool WebPagetest::RebootWatchDog() {
+  short reboots = LoadRebootWatchDog();
+  if (reboots >= MAX_REBOOT) {
+    return false;
+  }
+  SaveRebootWatchDog(reboots + 1);
+  return true;
+}
+
+/*-----------------------------------------------------------------------------
+Reset reboot count
+-----------------------------------------------------------------------------*/
+void WebPagetest::ResetRebootWatchDog() {
+  SaveRebootWatchDog(0);
+}
+
+void WebPagetest::SaveRebootWatchDog(short nb) {
+  CString val;
+  val.Format(_T("%d"), nb);
+  HANDLE file_handle = CreateFile(_T("reboot_count.txt"), GENERIC_WRITE, 0, 0,
+    CREATE_ALWAYS, 0, 0);
+  if (file_handle != INVALID_HANDLE_VALUE) {
+    DWORD bytes;
+    WriteFile(file_handle, val, val.GetLength(), &bytes, 0);
+    CloseHandle(file_handle);
+  }
+}
+
+short WebPagetest::LoadRebootWatchDog() {
+  byte ReadBuffer[1] = { 0 };
+
+  HANDLE file_handle = CreateFile(_T("reboot_count.txt"), GENERIC_READ, FILE_SHARE_READ,
+    0, OPEN_EXISTING, 0, 0);
+
+  DWORD bytes;
+  if (file_handle != INVALID_HANDLE_VALUE && ReadFile(file_handle, ReadBuffer, sizeof(ReadBuffer), &bytes, 0) && bytes) {
+    CloseHandle(file_handle);

```

Comment:
Strictly speaking, to be perfectly safe, I believe the CloseHandle should be moved out of the if statement (ReadFile can fail even thought file_handle is not invalid).